Examples/Algorithms/TrackFinding/src/TrackParamsEstimationAlgorithm.cpp

```diff
@@ -64,60 +73,130 @@ Acts::Vector3 ActsExamples::TrackParamsEstimationAlgorithm::getField(
       m_cfg.magneticField);
 }
 
+ActsExamples::SimSeedContainer
+ActsExamples::TrackParamsEstimationAlgorithm::createSeeds(
+    const ActsExamples::ProtoTrackContainer& protoTracks,
+    const ActsExamples::SimSpacePointContainer& spacePoints) const {
+  SimSeedContainer seeds;
+  seeds.reserve(protoTracks.size());
+  for (std::size_t itrack = 0; itrack < protoTracks.size(); ++itrack) {
+    // The list of hits and the initial start parameters
+    const auto& protoTrack = protoTracks[itrack];
+    if (protoTrack.size() < 3) {
+      ACTS_WARNING("Proto track " << itrack << " size is less than 3.");
+      continue;
+    }
+    // Space points on the proto track
+    std::vector<SimSpacePoint> spacePointsOnTrack;
+    spacePointsOnTrack.reserve(protoTrack.size());
+    // Loop over the hit index on the proto track to find the space points
+    for (const auto& hitIndex : protoTrack) {
+      auto it =
+          std::find_if(spacePoints.begin(), spacePoints.end(),
+                       [&](const SimSpacePoint& spacePoint) {
+                         return (spacePoint.measurementIndex() == hitIndex);
+                       });
+      if (it != spacePoints.end()) {
+        spacePointsOnTrack.push_back(*it);
+      }
+    }
+    // At least three space points are required
+    if (spacePointsOnTrack.size() < 3) {
+      continue;
+    }
+    // Sort the space points
+    std::sort(spacePointsOnTrack.begin(), spacePointsOnTrack.end(),
+              [](const SimSpacePoint& lhs, const SimSpacePoint& rhs) {
+                return lhs.r() < rhs.r();
+              });
+    // Loop over the found space points to find seeds with simple selection
+    for (size_t ib = 0; ib < spacePointsOnTrack.size(); ++ib) {
+      for (size_t im = ib + 1; im < spacePointsOnTrack.size(); ++im) {
+        for (size_t it = im + 1; it < spacePointsOnTrack.size(); ++it) {
```

Comment:
Here you could gain a couple of iteration by having `ib < spacePointsOnTrack.size()-2` and `im < spacePointsOnTrack.size()-1` but I am not sure it would make a big difference.

Reply:
Thanks! That makes sense. Changed.